apf/apfAggregateNumbering.h

```diff
@@ -0,0 +1,37 @@
+#ifndef APF_AGGREGATE_NUMBERING_H_
+#define APF_AGGREGATE_NUMBERING_H_
+#include "apfNew.h"
+#include <PCU.h>
+namespace apf
+{
+  class Field;
+  class FieldShape;
+  class Sharing;
+  class Mesh;
+  class MeshEntity;
+  template <class T>
+  class AggregateNumberingOf;
+  typedef AggregateNumberingOf<int> AggNumbering;
+  typedef AggregateNumberingOf<long> GlobalAggNumbering;
+  AggNumbering * createAggNumbering(Field * f,
+                                    int blocks,
+                                    int dofs_per_block,
+                                    MPI_Comm cm,
+                                    Sharing * share = NULL);
+  AggNumbering * createAggNumbering(Mesh * m,
+                                    const char * name,
+                                    FieldShape * shape,
+                                    int blocks,
+                                    int dofs_per_block,
+                                    MPI_Comm cm,
+                                    Sharing * share = NULL);
+  void destroyAggNumbering(AggNumbering * n);
+  int getElementNumbers(AggNumbering * n, MeshEntity * e, NewArray<int> & numbers);
```

Comment:
Ah but this needs an AggNumbering specifically rather than just a Numbering?

Reply:
Oh I might be able to just get rid of this since the underlying FieldData is now a ```UserData<int/long>``` which should operated correctly in the ```FieldDataOf<T>::getElementData()``` function I think.